core/src/connection/listeners.rs

```diff
@@ -283,12 +285,14 @@ where
                 Poll::Ready(None) => {
                     return Poll::Ready(ListenersEvent::Closed {
                         listener_id: *listener_project.id,
+                        addresses: listener.addresses.clone(),
```

Comment:
Since we're destroying the listener, you don't need to clone.
```suggestion
                        addresses: mem::replace(listener.addresses, SmallVec::new()),
```

(or, for `Vec`, `listener.addresses.drain(..).collect()`)

Reply:
I went with a `Vec` and used the `SmallVec` method `to_vec()` for the  conversion.